products/BellHybrid/apps/common/src/widgets/ListViewWithLabels.cpp

```diff
@@ -18,125 +23,133 @@ namespace gui
         body->dimensionChangedCallback = [&](gui::Item &, const BoundingBox &newDim) -> bool { return true; };
     }
 
-    void ListViewWithLabels::addItemsOnPage()
+    void ListViewWithLabels::addLabelItem()
     {
-        currentPageSize = 0;
-        itemsOnPage     = 0;
-        labelAdded      = false;
+        if (!labelsCount) {
+            return;
+        }
 
-        ListItem *item;
-        while ((item = provider->getItem(getOrderFromDirection())) != nullptr) {
-            /* If direction is top-to-bottom, add label mark before adding relaxation item. */
-            if (direction == listview::Direction::Bottom) {
-                addLabelMarker(item);
-            }
-            /* Check if new item fits, if it does - add it, if not - handle label insertion
-             * case for bottom-to-top navigation direction. */
-            if (getSlotsLeft() > 0) {
-                body->addWidget(item);
-                itemsOnPage++;
+        std::int32_t size = currentPageSize;
+        size              = direction == listview::Direction::Top ? size * -1 : size;
+
+        auto position = 0U;
+        for (std::uint32_t i = 0; i < labelsCount; ++i) {
```

Comment:
You can also use the notation with `U` specifier here so that it's consistent with the rest of this file:
```suggestion
        for (auto i = 0U; i < labelsCount; ++i) {
```

Reply:
Done